File: search-application/src/main/java/com/example/search/application/savedsearch/service/ListMySavedSearchesService.java
package com.example.search.application.savedsearch.service;

import com.example.search.application.savedsearch.port.in.ListMySavedSearchesUseCase;
import com.example.search.application.savedsearch.port.out.SavedSearchRepository;
import com.example.search.domain.savedsearch.SavedSearch;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
public class ListMySavedSearchesService implements ListMySavedSearchesUseCase {

    private final SavedSearchRepository repository;

    @Override
    @Transactional(readOnly = true)
    public List<SavedSearch> findByOwner(String ownerId) {
        return repository.findByOwner(ownerId);
    }
}

File: search-application/src/main/kotlin/com/example/search/application/savedsearch/service/ListMySavedSearchesService.kt
package com.example.search.application.savedsearch.service

import com.example.search.application.savedsearch.port.`in`.ListMySavedSearchesUseCase
import com.example.search.application.savedsearch.port.out.SavedSearchRepository
import com.example.search.domain.savedsearch.SavedSearch
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class ListMySavedSearchesService(
    private val repository: SavedSearchRepository
) : ListMySavedSearchesUseCase {

    @Transactional(readOnly = true)
    override fun findByOwner(ownerId: String): List<SavedSearch> {
        return repository.findByOwner(ownerId)
    }
}
